src/librustc_mir/shim.rs

```diff
@@ -66,9 +66,11 @@ fn make_shim<'tcx>(tcx: TyCtxt<'tcx>, instance: ty::InstanceDef<'tcx>) -> &'tcx
                 Some(arg_tys)
             )
         }
-        ty::InstanceDef::Virtual(def_id, _) => {
-            // We are generating a call back to our def-id, which the
-            // codegen backend knows to turn to an actual virtual call.
+        // We are generating a call back to our def-id, which the
+        // codegen backend knows to turn to an actual virtual call.
+        ty::InstanceDef::Virtual(def_id, _) |
+        // ...or we are generating a call to the inner closure defined by #[track_caller]
```

Comment:
"inner closure" sounds like the RFC text (but it's not what's going on in this implementation).

Maybe just keep one comment and change the "virtual" to "virtual/direct"? Not super sure here.